wasp/benchspy_test.go

```diff
@@ -0,0 +1,245 @@
+package wasp_test
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/wasp"
+	"github.com/smartcontractkit/chainlink-testing-framework/wasp/benchspy"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBenchSpyWithLokiQuery(t *testing.T) {
+	label := "benchspy"
+
+	gen, err := wasp.NewGenerator(&wasp.Config{
+		T:          t,
+		LokiConfig: wasp.NewEnvLokiConfig(),
+		GenName:    "vu",
+		Labels: map[string]string{
+			"branch": label,
+			"commit": label,
+		},
+		CallTimeout: 100 * time.Millisecond,
+		LoadType:    wasp.VU,
+		Schedule: wasp.CombineAndRepeat(
+			2,
+			wasp.Steps(10, 1, 10, 10*time.Second),
+			wasp.Plain(30, 15*time.Second),
+			wasp.Steps(20, -1, 10, 5*time.Second),
+		),
+		VU: wasp.NewMockVU(&wasp.MockVirtualUserConfig{
+			CallSleep: 50 * time.Millisecond,
+		}),
+	})
+	require.NoError(t, err)
+
+	currentReport := benchspy.StandardReport{
+		BasicData: benchspy.BasicData{
+			GeneratorConfigs: map[string]*wasp.Config{
+				gen.Cfg.GenName: gen.Cfg,
+			},
+			TestName:    t.Name(),
+			TestStart:   time.Now(),
+			CommitOrTag: "e7fc5826a572c09f8b93df3b9f674113372ce925",
+		},
+		ResourceReporter: benchspy.ResourceReporter{
+			ExecutionEnvironment: benchspy.ExecutionEnvironment_Docker,
+		},
+	}
+
+	lokiQueryExecutor := benchspy.NewLokiQueryExecutor(
+		map[string]string{
+			"vu_over_time": fmt.Sprintf("max_over_time({branch=~\"%s\", commit=~\"%s\", go_test_name=~\"%s\", test_data_type=~\"stats\", gen_name=~\"%s\"} | json | unwrap current_instances [10s]) by (node_id, go_test_name, gen_name)", label, label, t.Name(), gen.Cfg.GenName),
+		},
+		gen.Cfg.LokiConfig)
+
+	currentReport.QueryExecutors = append(currentReport.QueryExecutors, lokiQueryExecutor)
+
+	gen.Run(true)
+	currentReport.TestEnd = time.Now()
+
+	fetchCtx, cancelFn := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancelFn()
+
+	fetchErr := currentReport.FetchData(fetchCtx)
+	require.NoError(t, fetchErr, "failed to fetch current report")
+
+	// path, storeErr := currentReport.Store()
+	// require.NoError(t, storeErr, "failed to store current report", path)
+
+	previousReport := benchspy.StandardReport{
```

Comment:
Ideally, we hide it under the hood, if report comparison mode is set to last commit we find such report, unmarshal it and create this structure under the hood

Reply:
I modified the interfaces a bit to make that possible